core/src/main/scala/akka/kafka/KafkaConsumerActor.scala

```diff
@@ -74,7 +74,7 @@ private[kafka] class KafkaConsumerActor[K, V](settings: ConsumerSettings[K, V])
   val pollTask: Cancellable =
     context.system.scheduler.schedule(pollInterval(), pollInterval(), self, Poll)(context.dispatcher)
 
-  var requests = Map.empty[TopicPartition, ActorRef]
+  var requests = Map.empty[ActorRef, RequestMessages]
```

Comment:
@13h3r yes, isn't that the point with this shared actor? It's flattened in the `poll` and then the results are sent to the requestors. What do you mean by checking for existing requests?